rust/src/web/graphql/stakes/mod.rs

```diff
@@ -52,6 +53,17 @@ impl StakeQueryRoot {
         let mut accounts: Vec<LedgerAccountWithMeta> = staking_ledger
             .staking_ledger
             .into_values()
+            .filter(|account| {
+                if let Some(ref query) = query {
+                    if let Some(public_key) = &query.public_key {
+                        *public_key == account.pk.0
```

Comment:
why not `return *public_key == account.pk.0` and drop the `else`s?

Reply:
if the public key isn't set you don't want to filter